nixos/modules/system/boot/initrd-ssh.nix

```diff
@@ -134,20 +134,15 @@ in
       ${cfg.extraConfig}
     '';
   in mkIf (config.boot.initrd.network.enable && cfg.enable) {
+    warnings =
+      if cfg.hostKeys == []
```

Comment:
you can simplify the if statement with:
lib.optional (cfg.hostKeys == []) "..."

Reply:
Personally I am undecided whether thats more readable, but changed it as recommended.